src/observable/ConnectableObservable.ts

```diff
@@ -123,9 +129,11 @@ class RefCountSubscriber<T> extends Subscriber<T> {
     }
 
     const { connection } = this;
-    if (connection) {
-      this.connection = null;
-      connection.unsubscribe();
+    const sharedConnection = (<any> connectable)._connection;
+    this.connection = null;
+
+    if (sharedConnection && (!connection || sharedConnection === connection)) {
```

Comment:
Can you add a comment explaining this line? I don't want to grok it again later. haha.